corehq/apps/accounting/static/accounting/ko/accounting.stripe_card_manager.js

```diff
@@ -0,0 +1,153 @@
+            cardManager.wrap(data);
+            self.reset();
+        }).fail(function(data){
+            var response = JSON.parse(data.responseText);
+            self.errorMsg(response.error);
+        }).always(function(){
+            self.isProcessing(false);
+        });
+    };
+
+    var handleStripeResponse = function(status, response){
+        if (response.error){
+            self.isProcessing(false);
+            self.errorMsg(response.error.message);
+        } else {
+            self.errorMsg('');
+            self.token(response.id);
+            submit();
+        }
+    };
+
+    var createStripeToken = function(){
+        Stripe.card.createToken({
+            number: self.number(),
+            cvc: self.cvc(),
+            exp_month: self.expMonth(),
+            exp_year: self.expYear()
+        }, handleStripeResponse);
+    };
+
+    self.saveCard = function(){
+        self.isProcessing(true);
+        createStripeToken();
+    };
+};
+
+var StripeCard = function(card, baseUrl){
+    'use strict';
+    var self = this;
+    var mapping = {
+        include: ['brand', 'last4', 'exp_month','exp_year', 'is_autopay'],
+        copy: ['url', 'token']
+    };
+
+    self.wrap = function(data){
+        ko.mapping.fromJS(data, mapping, self);
+        self.is_autopay(self.is_autopay() == 'True');
+        self.url = baseUrl + card.token + '/';
+    };
+    self.wrap(card);
+
+    self.setAutopay = function(){
+        cardManager.autoPayButtonEnabled(false);
+        self.submit({is_autopay: true}).always(function(){
+            cardManager.autoPayButtonEnabled(true);
+        });
+    };
+
+    self.unSetAutopay = function(){
+        cardManager.autoPayButtonEnabled(false);
+        self.submit({is_autopay: false}).always(function(){
+            cardManager.autoPayButtonEnabled(true);
+        });
+    };
+
+    self.showDeleteConfirmation = ko.observable(false);
+    self.toggleDeleteConfirmation = function(){
+        self.showDeleteConfirmation(!self.showDeleteConfirmation());
+    };
+
+    self.deleteCard = function(card){
+        cardManager.cards.destroy(card);
+        $.ajax({
+            type: "DELETE",
+            url: self.url
+        }).success(function(data){
+            cardManager.wrap(data);
+        }).fail(function(data){
+            var response = JSON.parse(data.responseText);
+            alert_user(response.error, "error");
+            if (response.cards){
+                cardManager.wrap(response);
+            }
+        });
+    };
+
+    self.submit = function(data){
+        return $.ajax({
+            type: "POST",
+            url: self.url,
+            data: data
+        }).success(function(data){
+            cardManager.wrap(data);
+        }).fail(function(data){
+            var response = JSON.parse(data.responseText);
+            alert(response.error);
+        });
+    };
+};
+
+
+var StripeCardManager = function(data){
+    'use strict';
```

Comment:
in the future you can wrap all these in an anonymous function and just call `use strict` once i believe